File: C#/PrimeNumberGeneratorTests/MainProgramTests.cs
using System;
using System.IO;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PrimeNumberGeneratorExercise;

namespace PrimeNumberGeneratorTests
{
    [TestClass]
    public class MainProgramTests
    {
        const string USAGE = "Usage: PrimeNumberGenerator.exe <start value> <end value>\r\n<start value> and <end value> must be valid integers\r\n";

        [TestMethod]
        public void HappyPath_PrimeNumbersExistInRange()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "1", "10" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"Prime numbers in specified range [1, 10] (in ascending order):{Environment.NewLine}2 3 5 7 {Environment.NewLine}", sw.ToString());
        }

        [TestMethod]
        public void HappyPath_PrimeNumbersExistInRange_ReverseParameters()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "10", "1" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"Prime numbers in specified range [10, 1] (in ascending order):{Environment.NewLine}2 3 5 7 {Environment.NewLine}", sw.ToString());
        }

        [TestMethod]
        public void Negative_PrimeNumbersDoNotExistInRange()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "8", "10" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"No prime numbers exist in specified range [8, 10]{Environment.NewLine}", sw.ToString());
        }

        [TestMethod]
        public void Negative_PrimeNumbersDoNotExistInRange_ReverseParameters()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "10", "8" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"No prime numbers exist in specified range [10, 8]{Environment.NewLine}", sw.ToString());
        }

        [TestMethod]
        public void Negative_NegativeNumbersEntered()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "-1", "-5" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"No prime numbers exist in specified range [-1, -5]{Environment.NewLine}", sw.ToString());
        }

        [TestMethod]
        public void Negative_InvalidInteger_StartValue()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "abc", "5" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"One or more values is invalid{Environment.NewLine}{USAGE}", sw.ToString());
        }

        [TestMethod]
        public void Negative_InvalidInteger_EndValue()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "1", "abc" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"One or more values is invalid{Environment.NewLine}{USAGE}", sw.ToString());
        }

        [TestMethod]
        public void Negative_InvalidIntegers_BothValues()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "abc", "def" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"One or more values is invalid{Environment.NewLine}{USAGE}", sw.ToString());
        }

        [TestMethod]
        public void Negative_NotEnoughParameters()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "1" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"Please specify 2 integers{Environment.NewLine}{USAGE}", sw.ToString());
        }

        [TestMethod]
        public void Negative_TooManyParameters()
        {
            StringWriter sw = new StringWriter();
            Console.SetOut(sw);

            Program.Main(new[] { "1", "2", "3" });

            Assert.IsNotNull(sw);
            Assert.AreEqual($"Please specify 2 integers{Environment.NewLine}{USAGE}", sw.ToString());
        }
    }
}

File: Java/PrimeNumberGeneratorExercise/test/com/png/implementations/tests/mainProgramTests.java
package com.png.implementations.tests;
import java.io.*;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import com.png.*;

class mainProgramTests {
	static String USAGE = "Usage: PrimeNumberGenerator.exe <start value> <end value>\r\n<start value> and <end value> must be valid integers\r\n";
	
	@Test
	void happyPath_PrimeNumbersExistInRange() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "1", "10" });
		
		assertEquals("List of prime numbers in specified range [1, 10]:\r\n2 3 5 7 ", os.toString());
	}

	@Test
	void happyPath_PrimeNumbersExistInRange_ReverseParameters() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "10", "1" });
		
		assertEquals("List of prime numbers in specified range [10, 1]:\r\n2 3 5 7 ", os.toString());
	}
	
	@Test
	void negative_PrimeNumbersDoNotExistInRange() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "8", "10" });
		
		assertEquals("No prime numbers exist in specified range [8, 10]\r\n", os.toString());
	}
	
	@Test
	void negative_PrimeNumbersDoNotExistInRange_ReverseParameters() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "10", "8" });
		
		assertEquals("No prime numbers exist in specified range [10, 8]\r\n", os.toString());
	}
	
	@Test
	void negative_NegativeNumbersEntered() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "-1", "-5" });
		
		assertEquals("No prime numbers exist in specified range [-1, -5]\r\n", os.toString());
	}
	
	@Test
	void negative_InvalidInteger_StartValue() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "abc", "5" });
		
		assertEquals("One or more inputs is invalid\r\n" + USAGE, os.toString());
	}
	
	@Test
	void negative_InvalidInteger_EndValue() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "1", "abc" });
		
		assertEquals("One or more inputs is invalid\r\n" + USAGE, os.toString());
	}
	
	@Test
	void negative_InvalidIntegers_BothValues() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "abc", "def" });
		
		assertEquals("One or more inputs is invalid\r\n" + USAGE, os.toString());
	}
	
	@Test
	void negative_NotEnoughParameters() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "1" });
		
		assertEquals("Please specify 2 integers\r\n" + USAGE, os.toString());
	}
	
	@Test
	void negative_TooManyParameters() {
		OutputStream os = new ByteArrayOutputStream();
		PrintStream ps = new PrintStream(os);
		System.setOut(ps);
		
		PrimeNumberGeneratorExec.main(new String[] { "1", "2", "3" });
		
		assertEquals("Please specify 2 integers\r\n" + USAGE, os.toString());
	}
}
